src/main/java/io/jenkins/plugins/sprp/PipelineSnippetGenerator.java

```diff
@@ -126,18 +140,84 @@ public String getPublishReportSnippet(ArrayList<String> paths){
+        String snippet;
+        Descriptor stepDescriptor = StepDescriptor.byFunctionName(step.getStepName());
+
+        if(stepDescriptor == null)
+            throw new RuntimeException(new IllegalStateException("No step exist with the name of " + step.getStepName()));
+
+        Class clazz = stepDescriptor.clazz;
+
+        Object stepObject = null;
+
+        // Right now all the DefaultParameter of a step are considered to be string.
+        if(step.getDefaultParameter() != null){
+            Constructor c = clazz.getConstructor(String.class);
+            stepObject = c.newInstance(step.getDefaultParameter());
+        }
+        else{
+            Mapping mapping = new Mapping();
+
+            for(Map.Entry<String, Object> entry: step.getParameters().entrySet()){
+                Class stepFieldClass = clazz.getDeclaredField(entry.getKey()).getType();
+
+                if(stepFieldClass == String.class)
+                    mapping.put(entry.getKey(), (String) entry.getValue());
+                else if(stepFieldClass == Boolean.class)
+                    mapping.put(entry.getKey(), (Boolean) entry.getValue());
+                else if(stepFieldClass == Float.class)
+                    mapping.put(entry.getKey(), (Float) entry.getValue());
+                if(stepFieldClass == Double.class)
+                    mapping.put(entry.getKey(), (Double) entry.getValue());
+                else if(stepFieldClass == Integer.class)
+                    mapping.put(entry.getKey(), (Integer) entry.getValue());
+                else
+                    logger.log(Level.WARNING, stepFieldClass.getName() + "is not supported at this time.");
+            }
+
+            Configurator configurator = Configurator.lookup(clazz);
+            if (configurator != null) {
+                stepObject = configurator.configure(mapping);
+            }
+            else{
+                throw new IllegalStateException("No step with name '" + step.getStepName() +
+                        "' exist. Have you installed required plugin.");
+            }
+        }
+
+        snippet = Snippetizer.object2Groovy(stepObject) + "\n";
+        return snippet;
+    }
+
     public String getStage(
             Stage stage,
             ArrayList<String> buildResultPaths,
             ArrayList<String> testResultPaths,
             ArrayList<String> archiveArtifacts,
             GitConfig gitConfig,
             String findbugs
-    ){
+    ) throws NoSuchMethodException, InstantiationException, IllegalAccessException, ConfiguratorException,
+            InvocationTargetException, NoSuchFieldException
+    {
         String snippet = "stage('" + stage.getName() + "') {\n";
 
         snippet += "\tsteps {\n";
-        snippet += "\t\t" + addTabs(shellScript(stage.getScripts()), 2);
+        // Todo: needs to be fixed now only
+        snippet += "\t\t" + addTabs(getSteps(stage.getSteps()), 2);
         snippet += "\t}\n";
 
         if(stage.getFailure() != null
```

Comment:
In English, what does this long conditional mean? This would be worth a short comment perhaps.